persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStore.java

```diff
@@ -160,6 +183,66 @@ public void start() {
       }
    }
 
+   private void migrateFromV11() throws SQLException {
+      // If a custom user marshaller was previously used, no need to update rows
+      if (ctx.getGlobalConfiguration().serialization().marshaller() != null)
+         return;
+
+      Connection conn = null;
+      PreparedStatement ps = null;
+      ResultSet rs = null;
+      try {
+         conn = connectionFactory.getConnection();
+         conn.setAutoCommit(false);
+         String sql = tableManager.getLoadNonExpiredAllRowsSql();
+         ps = conn.prepareStatement(sql);
+         ps.setLong(1, timeService.wallClockTime());
+         rs = ps.executeQuery();
+
+         Marshaller userMarshaller = marshaller.getUserMarshaller();
+         try (PreparedStatement upsertBatch = conn.prepareStatement(tableManager.getUpdateRowSql())) {
+            int batchSize = 0;
+            while (rs.next()) {
+               batchSize++;
+               InputStream inputStream = rs.getBinaryStream(1);
+               String keyStr = rs.getString(2);
+               long timestamp = rs.getLong(3);
+               int segment = keyPartitioner == null ? -1 : rs.getInt(4);
+
+               MarshalledValue mv = unmarshall(inputStream, marshaller);
```

Comment:
I wonder should we confirm if the prior marshaller could be used? What if a DBA deletes the metadata table. I wonder if we should catch the exception and maybe mention that as a possibility?

Reply:
We could, however this will only be the case for Infinispan 12. From Infinispan 13 I think we should throw an exception if `!createOnStart` and the meta table doesn't exist as it's not possible to distinguish between 11 and 12 data if no meta table exists.